cloudant-spark-sql/src/main/scala/com/cloudant/spark/common/FilterUil.scala

```diff
@@ -134,7 +134,9 @@ class FilterUtil(filters: Map[String, Array[Filter]]){
           false
         }else
         {
-          if (field.isInstanceOf[JsNumber])  satisfiesAll(field.as[JsNumber].value.intValue(), filters)
+          if (field.isInstanceOf[JsNumber])  satisfiesAll(field.as[JsNumber].value.longValue(), filters)
+          else if (field.isInstanceOf[JsNumber])  satisfiesAll(field.as[JsNumber].value.intValue(), filters)
```

Comment:
The control flow will never go to line 138, because if field.isInstanceOf[JsNumber] it will always go to 137.